live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/mapper/property/PropertyMappingHelper.java

```diff
@@ -183,4 +187,26 @@ private static String getUserId(NodeResource node, Function<NodeResource, UserIn
                 .map(UserInfo::getId)
                 .orElse(null);
     }
+
+    private static String generateDigest(String content)
+    {
+        try
+        {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            byte[] hash = digest.digest(content.getBytes(StandardCharsets.UTF_8));
+            return Base64.getEncoder().encodeToString(hash);
+        }
+        catch (NoSuchAlgorithmException e)
+        {
+            throw new RuntimeException("Error generating digest", e);
+        }
+    }
+
+    private static Optional<String> getContentDigest(NodeResource nodeResource)
+    {
+        return ofNullable(nodeResource)
+                .map(NodeResource::getContent)
+                .map(ContentInfo::getMimeType)
+                .map(PropertyMappingHelper::generateDigest);
```

Comment:
If I've read this correctly then I think we're generating the digest of the mimetype string. This is a good start, but we need to use something that uniquely identifies the piece of content. My suggestion in the ticket is to include the node id, the property name (`cm:content`) and the version number.